include/libopencm3/nrf5/common/gpio_common.h

```diff
@@ -0,0 +1,966 @@
+#define GPIO_OUT_PIN17_High              (1 << 17)
+#define GPIO_OUT_PIN18_LOW               (0 << 18)
+#define GPIO_OUT_PIN18_Low               (0 << 18)
+#define GPIO_OUT_PIN18_HIGH              (1 << 18)
+#define GPIO_OUT_PIN18_High              (1 << 18)
+#define GPIO_OUT_PIN19_LOW               (0 << 19)
+#define GPIO_OUT_PIN19_Low               (0 << 19)
+#define GPIO_OUT_PIN19_HIGH              (1 << 19)
+#define GPIO_OUT_PIN19_High              (1 << 19)
+#define GPIO_OUT_PIN20_LOW               (0 << 20)
+#define GPIO_OUT_PIN20_Low               (0 << 20)
+#define GPIO_OUT_PIN20_HIGH              (1 << 20)
+#define GPIO_OUT_PIN20_High              (1 << 20)
+#define GPIO_OUT_PIN21_LOW               (0 << 21)
+#define GPIO_OUT_PIN21_Low               (0 << 21)
+#define GPIO_OUT_PIN21_HIGH              (1 << 21)
+#define GPIO_OUT_PIN21_High              (1 << 21)
+#define GPIO_OUT_PIN22_LOW               (0 << 22)
+#define GPIO_OUT_PIN22_Low               (0 << 22)
+#define GPIO_OUT_PIN22_HIGH              (1 << 22)
+#define GPIO_OUT_PIN22_High              (1 << 22)
+#define GPIO_OUT_PIN23_LOW               (0 << 23)
+#define GPIO_OUT_PIN23_Low               (0 << 23)
+#define GPIO_OUT_PIN23_HIGH              (1 << 23)
+#define GPIO_OUT_PIN23_High              (1 << 23)
+#define GPIO_OUT_PIN24_LOW               (0 << 24)
+#define GPIO_OUT_PIN24_Low               (0 << 24)
+#define GPIO_OUT_PIN24_HIGH              (1 << 24)
+#define GPIO_OUT_PIN24_High              (1 << 24)
+#define GPIO_OUT_PIN25_LOW               (0 << 25)
+#define GPIO_OUT_PIN25_Low               (0 << 25)
+#define GPIO_OUT_PIN25_HIGH              (1 << 25)
+#define GPIO_OUT_PIN25_High              (1 << 25)
+#define GPIO_OUT_PIN26_LOW               (0 << 26)
+#define GPIO_OUT_PIN26_Low               (0 << 26)
+#define GPIO_OUT_PIN26_HIGH              (1 << 26)
+#define GPIO_OUT_PIN26_High              (1 << 26)
+#define GPIO_OUT_PIN27_LOW               (0 << 27)
+#define GPIO_OUT_PIN27_Low               (0 << 27)
+#define GPIO_OUT_PIN27_HIGH              (1 << 27)
+#define GPIO_OUT_PIN27_High              (1 << 27)
+#define GPIO_OUT_PIN28_LOW               (0 << 28)
+#define GPIO_OUT_PIN28_Low               (0 << 28)
+#define GPIO_OUT_PIN28_HIGH              (1 << 28)
+#define GPIO_OUT_PIN28_High              (1 << 28)
+#define GPIO_OUT_PIN29_LOW               (0 << 29)
+#define GPIO_OUT_PIN29_Low               (0 << 29)
+#define GPIO_OUT_PIN29_HIGH              (1 << 29)
+#define GPIO_OUT_PIN29_High              (1 << 29)
+#define GPIO_OUT_PIN30_LOW               (0 << 30)
+#define GPIO_OUT_PIN30_Low               (0 << 30)
+#define GPIO_OUT_PIN30_HIGH              (1 << 30)
+#define GPIO_OUT_PIN30_High              (1 << 30)
+#define GPIO_OUT_PIN31_LOW               (0 << 31)
+#define GPIO_OUT_PIN31_Low               (0 << 31)
+#define GPIO_OUT_PIN31_HIGH              (1 << 31)
+#define GPIO_OUT_PIN31_High              (1 << 31)
```

Comment:
I don't see any value in these defines personally.  
They certainly don't need to be duplicated for GPIO_OUTSET below.